processor/logcountprocessor/processor.go

```diff
@@ -0,0 +1,177 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logcountprocessor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/observiq/observiq-otel-collector/receiver/routereceiver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// processor is a processor that counts logs.
+type processor struct {
+	config    *Config
+	matchExpr *Expression
+	attrExprs map[string]*Expression
+	counter   *LogCounter
+	consumer  consumer.Logs
+	logger    *zap.Logger
+	cancel    context.CancelFunc
+	mux       sync.Mutex
+}
+
+// newProcessor returns a new processor.
+func newProcessor(config *Config, consumer consumer.Logs, matchExpr *Expression, attrExprs map[string]*Expression, logger *zap.Logger) *processor {
+	return &processor{
+		config:    config,
+		matchExpr: matchExpr,
+		attrExprs: attrExprs,
+		counter:   NewLogCounter(),
+		consumer:  consumer,
+		logger:    logger,
+	}
+}
+
+// Start starts the processor.
+func (p *processor) Start(ctx context.Context, _ component.Host) error {
+	processorCtx, cancel := context.WithCancel(ctx)
+	p.cancel = cancel
+	go p.handleMetricInterval(processorCtx)
+
+	return nil
+}
+
+// Capabilities returns the consumer's capabilities.
+func (p *processor) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// Shutdown stops the processor.
+func (p *processor) Shutdown(_ context.Context) error {
+	p.cancel()
+	return nil
+}
+
+// ConsumeLogs processes the logs.
+func (p *processor) ConsumeLogs(ctx context.Context, pl plog.Logs) error {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	records := convertToRecords(pl)
+	for _, record := range records {
+		if p.matchRecord(record) {
+			resource := p.extractResource(record)
+			attrs := p.extractAttributes(record)
+			p.counter.Add(resource, attrs)
+		}
+	}
+
+	return p.consumer.ConsumeLogs(ctx, pl)
+}
+
+// handleMetricInterval sends metrics at the configured interval.
+func (p *processor) handleMetricInterval(ctx context.Context) {
+	ticker := time.NewTicker(p.config.Interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			p.sendMetrics(ctx)
+		}
+	}
+}
+
+// sendMetrics sends metrics to the consumer.
+func (p *processor) sendMetrics(ctx context.Context) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	metrics := p.createMetrics()
+	p.counter.Reset()
+
```

Comment:
```suggestion
	if metrics.MetricCount() == 0 {
		return
	}
```

Do you have any thoughts about not sending empty metrics when there is nothing to report? This can cause unnecessary exporter/processor overhead if there is nothing left to do.